Configurations/README

```diff
@@ -467,6 +467,10 @@ include paths the build of their source files should use:
 
     INCLUDE[foo]=include
 
+It's also possible to specify C macros that should be defined:
+
+    DEFINE[foo]=FOO BAR=1
```

Comment:
Does this add "-DFOO -DBAR=1" (or its equiv) to the cflag?  A statement like that would be helpful.

Reply:
Yes, please.  Maybe say what happens both for linux and windows at least. It might avoid getting asked.  What happens if multiple files specify the same flags with different values?

Reply:
Note that they get attached to product files, i.e files to be built, rather than source. But it's true that they can get attached to intermediary product files as well (i.e object files) and that might create conflict (you could end up with `-DFOO=0 -DBAR -DFOO=2012`). I could add code to mitigate that sort of thing...

Reply:
A remark, though: `build.info` files are platform agnostic, so everything in them is inherently applicable on all platforms. That's a tenet.

Reply:
Yes, which is why examples of what it means for the two most popular platforms might be worthwhile. Printing a warning message if you find conflicting values is a good idea.

Reply:
I've looked around a bit more to see if we provide examples of that sort in these documents at all...  and we don't.  *This* PR isn't about rewriting the READMEs to get better, *this* PR is about adding macro capability.

So with that in mind, are you satisfied enough if I say that I intend to rewrite these READMEs entirely, and have made a mental note of the request for concrete examples / results?

Reply:
Sure, thanks.